File: 2271-rearrange-array-elements-by-sign/2271-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int n = nums.length;
        int[] pos = new int[n];
        int[] neg = new int[n];
        int p = 0, q = 0;
        for (int num : nums) {
            if (num >= 0) pos[p++] = num;
            else neg[q++] = num;
        }
        int[] result = new int[n];
        int i = 0, pi = 0, ni = 0;
        while (pi < p && ni < q) {
            result[i++] = pos[pi++];
            result[i++] = neg[ni++];
        }
        while (pi < p) result[i++] = pos[pi++];
        while (ni < q) result[i++] = neg[ni++];
        return result;
    }
}

File: 2271-rearrange-array-elements-by-sign/2271-rearrange-array-elements-by-sign.py
class Solution(object):
    def rearrangeArray(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        answer = []
        positive = [x for x in nums if x>=0]
        neative = [x for x in nums if x<0]
        for i, j in zip(positive, neative):
            answer.extend([i, j])

        return answer
        